### DISCUSSION TITLE
After ESP32S3 uses UF2 to install micropython firmware, the file system reports error

### DISCUSSION THREAD

**Discussion by Wind-stormger** [micropython/micropython#9961] [Core Development]
I opened an issue on Mar 25 #8453.
Maybe I could come here to ask a question and discuss this.

Briefly:
1. Get a UF2 firmware that supports EPS32S3 from here [adafruit/tinyuf2/releases](https://github.com/adafruit/tinyuf2/releases), use esptool to install it (0x0 combined.bin).
2. Get a micropython firmware with .uf2 extension that supports EPS32S3 from here https://micropython.org/download/, copy the .uf2 file to UF2 BootLoader as you would with rp2040.
3. Use any tool that has access to the REPL, such as [mpremote](https://pypi.org/project/mpremote/), enter REPL, soft reset, get the following error:
```
>>>
MPY: soft reboot
Performing initial setup
Traceback (most recent call last):
  File "_boot.py", line 11, in <module>
  File "inisetup.py", line 36, in setup
OSError: [Errno 22] EINVAL
MicroPython v1.19.1 on 2022-11-14; ESP32S3 module (spiram) with ESP32S3
Type "help()" for more information.
>>>
```
The error message points to this piece of code, which is used to format the entire flash using littlefs v2.
https://github.com/micropython/micropython/blob/d9bca305e576a98192724c212c31413395d31cb4/ports/esp32/modules/inisetup.py#L33-L38

But the file system format used by UF2 is FAT, so here is error.

**Comment by Wind-stormger** [comment]
Solved by manually formatting to FAT filesystem, but could there be a better way?
```
>>> os.VfsFat.mkfs(bdev)
>>> os.mount(bdev,'/')
>>>
MPY: soft reboot
MicroPython v1.19.1 on 2022-11-14; ESP32S3 module (spiram) with ESP32S3
Type "help()" for more information.
>>>
```

**Comment by Wind-stormger** [comment]
Whenever people download micropython firmware, at the top of the list is the UF2 firmware.
<img width="362" alt="20230620153057" src="https://github.com/micropython/micropython/assets/76462385/fa1b9018-b7e9-405a-8a6c-80f8ee25d30c">

But in the UF2 Bootloader, if you have not installed the circuitpython.uf2 firmware before, but directly installed the micropython.uf2, this error must occur.

```py
Performing initial setup
Traceback (most recent call last):
  File "_boot.py", line 11, in <module>
  File "inisetup.py", line 40, in setup
OSError: [Errno 22] EINVAL
```

I want to add code that can format the partition as fat in ports/esp32/modules/inisetup.py.

https://github.com/micropython/micropython/blob/ca79b4961953be587a290bc995ae925e664c9fbc/ports/esp32/modules/inisetup.py#L37-L53

like this:

```py
def setup():
    from esp32 import Partition
    check_bootsec()
    print("Performing initial setup")
    if Partition.info(bdev)[4] == "vfs":
        os.VfsLfs2.mkfs(bdev)
        vfs = os.VfsLfs2(bdev)
    elif Partition.info(bdev)[4] == "ffat":
        os.VfsFat.mkfs(bdev)
        vfs = os.VfsFat(bdev)
    os.mount(vfs, "/")
    with open("boot.py", "w") as f:
        f.write(
            """\
# This file is executed on every boot (including wake-boot from deepsleep)
#import esp
#esp.osdebug(None)
#import webrepl
#webrepl.start()
"""
        )
    return vfs

```

@robert-hh , @jimmo  

Would you please，could I submit a PR on this?

**Comment by peterhinch** [comment]
The instructions suggest using `esptool.py`, with the command line specifying the `.bin` file. This has always worked perfectly for me.

**Reply by Wind-stormger** [reply]
Yes, I did that too.

I'm just trying to get .uf2 to work as well. 😃
